src/fn/pinrow.ts

```diff
@@ -9,6 +9,12 @@ export const pinrow_def = z.object({
   p: length.default("0.1in").describe("pitch"),
   id: length.default("1.0mm").describe("inner diameter"),
   od: length.default("1.5mm").describe("outer diameter"),
+  male: z.boolean().optional().default(true).describe("for male pin headers"),
```

Comment:
you have to use a transform to set `male` to true only if `female` isn't defined
```suggestion
  male: z.boolean().optional().describe("for male pin headers"),
```